exo-setup/src/app-setup.spec.ls

```diff
@@ -0,0 +1,123 @@
+require! {
+  '../../exosphere-shared': {run-process}
+  'chai': {expect}
+  'fs-extra' : fs
+  'js-yaml' : yaml
+  'mkdirp'
+  'path'
+  'rimraf'
+}
+
+
+describe 'AppSetup', ->
+
+  describe 'app with external dependencies' (...) ->
+    @timeout 600000
+
+    before (done) ~>
+      @app-name = 'external-dependency'
+      @app-dir = path.join process.cwd!, 'test', @app-name
+      @docker-compose-location = path.join @app-dir, 'tmp', 'docker-compose.yml'
+      @exocom-name = 'exocom0.22.1'
+      @service-name = 'mongo'
+      @mongo-name = 'mongo3.4.0' # external dependency
+      @process = checkout-setup-app @app-name, @app-dir
+        ..on 'ended' done
+
+    specify 'should create docker-compose.yml at the expected location', (done) ~>
+      fs.exists @docker-compose-location, (exists) ~>
+        expect(exists).to.be.true
+        @docker-compose = yaml.safe-load fs.read-file-sync(@docker-compose-location, 'utf8')
+        done!
+
+    specify 'should list all services and dependencies under \'services\'', (done) ~>
+      expect(Object.keys @docker-compose.services).to.eql([@exocom-name, @service-name, @mongo-name])
+      done!
+
+    specify 'should generate an image name for each dependency', (done) ~>
+      expect(@docker-compose.services[@exocom-name].image).to.not.be.empty
+      expect(@docker-compose.services[@mongo-name].image).to.not.be.empty
+      done!
+
+    specify 'should generate a container name for each service/dependency equal to the service/dependency name', (done) ~>
```

Comment:
Just `should generate a container name for each service and dependency` should be fine. Then we can check for equality in the steps below like you're already doing.